packages/data/src/plugins/persistence/index.js

```diff
@@ -252,29 +252,31 @@ export function migrateFeaturePreferencesToPreferencesStore(
 		: sourceFeatures;
 
 	if ( featuresToMigrate ) {
-		const targetFeatures = state[ preferencesStoreName ]?.preferences;
+		const existingPreferences = state[ preferencesStoreName ]?.preferences;
 
 		// Avoid migrating features again if they've previously been migrated.
-		if ( ! targetFeatures?.[ sourceStoreName ] ) {
+		if ( ! existingPreferences?.[ sourceStoreName ] ) {
 			// Set the feature values in the interface store, the features
 			// object is keyed by 'scope', which matches the store name for
 			// the source.
 			persistence.set( preferencesStoreName, {
 				preferences: {
-					...targetFeatures,
+					...existingPreferences,
```

Comment:
I think it can be `undefined`, but from what I can tell `undefined` spreads ok within objects:
```js
{ ...undefined, test: 1 }    // outputs `{ test: 1 }`
```

Only array spreads seem to need an iterable value:
```js
[ ...undefined, 1 ]     // throws an error - undefined is not iterable.
```

I thought I'd encountered past situations where it didn't work, so this explains it!

Reply:
Ah, perfect. Thanks for the explanation! I'll resolve all these comments 😀